### PR TITLE
initial multiplayer structure

### PR DESCRIPTION
**Pull request by aleasto**

### CODE REVIEW

**Comment by aleasto** [comment]
Don't merge this just yet

**Review by aleasto — COMMENTED** [review]
--- how to delete a review xd

File: super-cool-game/Assets/Scripts/Multiplayer/MultiplayerGame.cs [hunk 1]
```diff
@@ -15,9 +15,9 @@ public class MultiplayerGame : MonoBehaviour {
     }
 
     // Update is called once per frame
-    void Update() {
+    void FixedUpdate() {
         // Process UDP packets
-        if (NetCode.CmdQueue.Count > 0) {
+        while (NetCode.CmdQueue.Count > 0) {
```
Comment (aleasto) [inline super-cool-game/Assets/Scripts/Multiplayer/MultiplayerGame.cs]: This (lines 18-20) is kindof a crucial choice. I'm open to discussion.

Comment (aleasto) [reply super-cool-game/Assets/Scripts/Multiplayer/MultiplayerGame.cs]: It seems to me that without this, 2 remote clients are better synchronized

File: super-cool-game/Assets/Scripts/Player/PlayerController.cs [hunk 2]
```diff
@@ -1,33 +1,45 @@
...
 	}
 
     public void Move(float moveX) {
         var velX = Mathf.Max(-1, Mathf.Min(1, moveX)) * speed;
         if (this.velX != velX) {
-            pm.Move(velX);
+            pm.MoveX(velX);
             this.velX = velX;
-            // TODO: Send to network
+            if (NetCode.IsConnected) {
```
Comment (andreafra) [inline super-cool-game/Assets/Scripts/Player/PlayerController.cs:26]: Maybe group it up in a function?

**Review by andreafra — APPROVED** [review]

File: super-cool-game/Assets/Scripts/Player/PlayerController.cs (hunk 2 above)
Comment (aleasto) [reply super-cool-game/Assets/Scripts/Player/PlayerController.cs:26]: Looks tidy enough to me

**Pull request merged**